content-repo/gen_top_contrib.py

```diff
@@ -232,58 +255,96 @@ def get_inner_pr_request() -> list:
                 github_profile = user_info.get('html_url')
                 if not github_avatar and not github_profile:
-                    print(f'The user "{contributor}" was not found.')
+                    print(f'The user "{user}" was not found.')
                     continue
 
-                users.append({
-                    'Contributor': f"<img src='{github_avatar}'/><br></br> "
-                                   f"<a href='{github_profile}' target='_blank'>{contributor}</a>"
-                })
+        if user not in users:
+            users.update({
+                user: {
+                    'user': user,
+                    'github_profile': github_profile,
+                    'github_avatar': github_avatar,
+                    'number_of_contributions': 1
+                }
+            })
+        else:
+            users[user]['number_of_contributions'] += 1
 
-    for user in users:
-        prs = users.count(user)
-        user.update({'Number of Contribution(s)': prs})
+    if last_update:
+        users.update({'last_update': last_update})
 
-    list_users = []
-    result = {i['Contributor']: i for i in reversed(users)}.values()
-    new_res = sorted(result, key=lambda k: k['Number of Contribution(s)'], reverse=True)
+    return users
 
-    for user in new_res:
-        user['Contributor'] += f'<br></br>{user["Number of Contribution(s)"]} Contributions'
-        list_users.append(user['Contributor'])
 
-    return list_users
+def create_users_list(list_users: list) -> List:
+    users = []
+    contributors = []
+
+    for key, value in list_users:
+        users.append({
+            'Contributor': f"<img src='{value.get('github_avatar')}'/><br></br> "
+                           f"<a href='{value.get('github_profile')}' target='_blank'>{value.get('user')}</a>"
+                           f"<br></br>{value.get('number_of_contributions')} Contributions"
+        })
+
+    for contributor in users:
+        contributors.append(contributor['Contributor'])
+
+    return contributors
+
+
+def update_users_list(users_dict: dict, contributors_file_dict: dict):
+    """
+    Update the contributors list with the number of the current PRs
+    Args:
+        users_dict (dict): the users list
+        contributors_file_dict (dict): The list from the contributors file to update
+
+    Returns: The updated users dict - to update the contributors file
+
+    """
+    # In case of no new prs after last timestamp user_dict can be empty
+    if users_dict:
+        # Updating the last update timestamp
+        last_update = users_dict.pop('last_update')
+        contributors_file_dict['last_update'] = last_update
+        for user in users_dict.items():
+            new_prs = user[1]['number_of_contributions']
+            # Case of new user
+            if user[0] not in contributors_file_dict.keys():
+                contributors_file_dict.update({user[0]: user[1]})
```

Comment:
What is `user[0]` `user[1]`?

Reply:
@Shellyber what about this?

Reply:
Sorry, user[0] is the key - the contributor name
user[1] is the value of it - the dictionary containing all the data